### PR TITLE
S24/trinity/adding pet relationship tables

### PR DESCRIPTION
**Pull request by trinity-y**
## Notion ticket link

[Pet Relationship Tables](https://www.notion.so/uwblueprintexecs/Sprint-Board-4f2ffd2639ac41c4ad0f64818d84d5fa?p=c4bdff8ae86b4190abebd080bb13532a&pm=s)

## Implementation description
* added pet_behaviours table and user_pet_activities table
* scrapped pet_activities table for now, as the only use would be for automation of activities and that’s not a priority
* non-assigned tasks go into user_pet_activities where user column is null
* status in user_pet_activities is removed, not started/in progress/complete will be calculated from start and end timestamps

## Steps to test
- tested tables, checked for null values + foreign keys
image
image

## What should reviewers focus on?
* changed Activities class name to Activity, is this ok/should this change be put somewhere else?

### CODE REVIEW

File: backend/typescript/migrations/2024.07.08T19.19.43.create-user-pet-activity.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { DataType } from "sequelize-typescript";
+
+import { Migration } from "../umzug";
+
+const TABLE_NAME = "user_pet_activities";
+
+export const up: Migration = async ({ context: sequelize }) => {
+  await sequelize.getQueryInterface().createTable(TABLE_NAME, {
+    user_pet_activity_id: {
```
Comment (jerry-cheng5) [inline backend/typescript/migrations/2024.07.08T19.19.43.create-user-pet-activity.ts:9]: i think sequelize automatically creates an id field in the model, so you can just name this 'id', and you can remove the 'user_pet_activity_id' from the model as well

**Review by jerry-cheng5 — APPROVED** [review]
looks good to me! just a small change you could make to take advantage of what sequelize does for us

File: backend/typescript/migrations/2024.07.08T19.19.43.create-user-pet-activity.ts (hunk 1 above)
Comment (sthuray) [reply backend/typescript/migrations/2024.07.08T19.19.43.create-user-pet-activity.ts:9]: You're also missing an id for the pet-behaviours table

Comment (trinity-y) [reply backend/typescript/migrations/2024.07.08T19.19.43.create-user-pet-activity.ts:9]: there was no primary key in the pet behaviours table diagram so I left it out, I can add it if we need it?

**Review by aanxniee — APPROVED** [review]
good work! let's wait for daniel and lakshya to take a look at this

**Review by sthuray — APPROVED** [review]
Looks good!

**Pull request merged**